### PR TITLE
add comments to queryservice configs

### PR DESCRIPTION
**Pull request by yaoshengzhe**

### CODE REVIEW

File: go/vt/tabletserver/queryctl.go [hunk 1]
```diff
@@ -32,32 +32,32 @@ var (
...
+	flag.IntVar(&qsConfig.StreamPoolSize, "queryserver-config-stream-pool-size", DefaultQsConfig.StreamPoolSize, "query server stream pool size, stream pool is used by stream queries: queries that return results to client in a streaming fashion")
+	flag.IntVar(&qsConfig.TransactionCap, "queryserver-config-transaction-cap", DefaultQsConfig.TransactionCap, "query server transaction cap is the maximum number of transactions allowed to happen at any given point of a time for a single vttablet. E.g. by setting transaction cap to 100, there are at most 100 transactions will be processed by a vttablet and the 101th transaction will be blocked (and fail if it cannot get connection within specified timeout)")
+	flag.Float64Var(&qsConfig.TransactionTimeout, "queryserver-config-transaction-timeout", DefaultQsConfig.TransactionTimeout, "query server transaction timeout (in second), a transaction will be killed if it takes longer than this value")
+	flag.IntVar(&qsConfig.MaxResultSize, "queryserver-config-max-result-size", DefaultQsConfig.MaxResultSize, "query server max result size, maximum number of rows allowed to return from vttablet.")
```
Comment (sougou) [inline go/vt/tabletserver/queryctl.go]: "for non-streaming queries"

File: go/vt/tabletserver/queryctl.go [hunk 2]
```diff
@@ -32,32 +32,32 @@ var (
...
+	flag.IntVar(&qsConfig.StreamBufferSize, "queryserver-config-stream-buffer-size", DefaultQsConfig.StreamBufferSize, "query server stream buffer size, the maximum number of bytes sent from vttablet for each stream call.")
+	flag.IntVar(&qsConfig.QueryCacheSize, "queryserver-config-query-cache-size", DefaultQsConfig.QueryCacheSize, "query server query cache size, maximum number of queries to be cached. vttablet analyzes every incoming query and generate a query plan, these plans are being cached in a lru cache. This config controls the capacity of the lru cache.")
+	flag.Float64Var(&qsConfig.SchemaReloadTime, "queryserver-config-schema-reload-time", DefaultQsConfig.SchemaReloadTime, "query server schema reload time, how often vttablet reloads schemas from underlying MySQL instance in second. vttablet keeps table schemas in its own memory and periodically refresh it from MySQL. This config controls the reload time.")
```
Comment (sougou) [inline go/vt/tabletserver/queryctl.go]: refreshes

Comment (sougou) [inline go/vt/tabletserver/queryctl.go]: seconds (need to correct all places).

File: go/vt/tabletserver/queryctl.go [hunk 3]
```diff
@@ -32,32 +32,32 @@ var (
...
+	flag.Float64Var(&qsConfig.TxPoolTimeout, "queryserver-config-txpool-timeout", DefaultQsConfig.TxPoolTimeout, "query server transaction pool timeout, it is how long vttablet waits if tx pool is full")
+	flag.Float64Var(&qsConfig.IdleTimeout, "queryserver-config-idle-timeout", DefaultQsConfig.IdleTimeout, "query server idle timeout (in second), vttablet manages various mysql connection pools. This config means if a connection has not been used in given idle timeout, this connection will be removed from pool. This effectively manages number of connection objects and optimize the pool performance.")
+	flag.Float64Var(&qsConfig.SpotCheckRatio, "queryserver-config-spot-check-ratio", DefaultQsConfig.SpotCheckRatio, "query server rowcache spot check frequency (in [0, 1]), vttablet returns query result from rowcache (if is enabled) if they hits the cache; however, vttablet sometimes also sends the same query to MySQL to do a double check even if it hits the rowcache. In other words, this config controls how frequent of such checks.")
```
Comment (sougou) [inline go/vt/tabletserver/queryctl.go]: If rowcache is enabled, this value determines how often a row retrieved from the rowcache is spot-checked against MySQL.

File: go/vt/tabletserver/queryctl.go [hunk 4]
```diff
@@ -32,32 +32,32 @@ var (
...
 	flag.BoolVar(&qsConfig.StrictMode, "queryserver-config-strict-mode", DefaultQsConfig.StrictMode, "allow only predictable DMLs and enforces MySQL's STRICT_TRANS_TABLES")
 	flag.BoolVar(&qsConfig.StrictTableAcl, "queryserver-config-strict-table-acl", DefaultQsConfig.StrictTableAcl, "only allow queries that pass table acl checks")
 	flag.BoolVar(&qsConfig.TerseErrors, "queryserver-config-terse-errors", DefaultQsConfig.TerseErrors, "prevent bind vars from escaping in returned errors")
-	flag.StringVar(&qsConfig.RowCache.Binary, "rowcache-bin", DefaultQsConfig.RowCache.Binary, "rowcache binary file")
+	flag.StringVar(&qsConfig.RowCache.Binary, "rowcache-bin", DefaultQsConfig.RowCache.Binary, "rowcache binary file, vttablet launches a memcached if rowcache is enabled. This config specifies the location of the memecache binary.")
```
Comment (sougou) [inline go/vt/tabletserver/queryctl.go]: memecache -> memcache

**Comment by sougou** [comment]
lgtm.
Good work. Some minor comments.

**Comment by yaoshengzhe** [comment]
fixed.

**Pull request merged**